src/Workspaces/Core/Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -276,17 +280,50 @@ public LoadState(IReadOnlyDictionary<string, ProjectId> projectPathToProjectIdMa
             public void Add(ProjectInfo info)
             {
                 _projectIdToProjectInfoMap.Add(info.Id, info);
-                _projectInfoList.Add(info);
+                //Memoized results of ProjectAlreadyReferencesProject may no longer be correct;
+                //reset the cache.
+                ProjectAlreadyReferencesProjectMemo = new Dictionary<ProjectId, Dictionary<ProjectId, bool>>();
             }
 
-            public bool TryGetValue(ProjectId id, out ProjectInfo info)
+            private bool TryGetValue(ProjectId id, out ProjectInfo info)
```

Comment:
Because it's no longer used externally to the class and my bias is to limit public surface area whenever possible. Frankly, as it's only used once within the class and is a pure wrapper, I should probably have nuked it entirely. Will do for next commit.